vm.go

```diff
@@ -262,6 +262,9 @@ func floatToInt(f float64) (result int64, ok bool) {
 }
 
 func floatToValue(f float64) (result Value) {
+	if i, ok := floatToInt(f); ok {
+		return intToValue(i)
+	}
```

Comment:
Somehow we lost this during some updates, it checks if a float value can be converted to an integer and if so it becomes `valueInt` and it's handled as such everywhere else.

Reply:
Spoke too soon, I wish we documented this change. If I keep this in then roundtrips with ejson will "lose" the type (e.g. sending a float and getting an int back from ejson instead, since here we try to convert it to int if possible).